lib/matplotlib/tests/test_subplots.py

```diff
@@ -212,3 +212,22 @@ def test_dont_mutate_kwargs():
 def test_subplot_factory_reapplication():
     assert maxes.subplot_class_factory(maxes.Axes) is maxes.Subplot
     assert maxes.subplot_class_factory(maxes.Subplot) is maxes.Subplot
+
+
+@check_figures_equal(extensions=['png'])
+def test_width_and_height_ratios(fig_test, fig_ref):
+    fig_test.subplots(2, 3, height_ratios=[1, 2],
+                      width_ratios=[1, 3, 2])
+    fig_ref.subplots(2, 3, gridspec_kw={
+                     'height_ratios': [1, 2],
+                     'width_ratios': [1, 3, 2]})
+
```

Comment:
my personal nit (not blocking) preference is that height and width should be checked independently

Reply:
Sure, probably not a bad idea to test None...